### PR TITLE
Add documentation for RouteSnapshots/ flushing

### PR DESCRIPTION
**Pull request by loganakamatsu**

### CODE REVIEW

File: source/includes/getting-started.md [hunk 1]
```diff
@@ -817,6 +817,59 @@ In order to begin the second Route, you'll need to finish the first one by makin
 When all the routes have been finished, your [RouteStateListener](https://foxtrotsystems.github.io/android-sdk-javadoc/io/foxtrot/android/sdk/state/RouteStateListener.html) will receive a call to `onAllRoutesFinished()` and you'll have the opportunity to respond.
 
 
+##Flushing Route Activity
+
+The Foxtrot SDK is designed to work without a consistent internet connection. As a consequence, the route activity and data collected during the route must be completely synced to the server when the route is finished. This route activity is encapsulated in an [RouteSnapshot]() object. RouteSnapshots can be obtained by by calling [FinishRoute]().
```
Comment (raimille1) [inline source/includes/getting-started.md]: RouteSnapshots are created after calling [FinishRoute]() and once synced with our servers you will receive a call to onUploadComplete().

File: source/includes/getting-started.md [hunk 2]
```diff
@@ -817,6 +817,59 @@ In order to begin the second Route, you'll need to finish the first one by makin
...
+    String snapshotId = routeSnapshot.getId();   // This will match the ID from the snapshot returned from FoxtrotSDK.finishRoute()
+    String routeId = routeSnapshot.getRouteId(); // This is the same id as on OptimizedRoute.getId() 
+  }
+}
+```
+
+You can then listen for routes getting flushed as follows:
+
+```java
+  YourRouteSnapshotSyncListener yourRouteSnapshotSyncListener = ...
```
Comment (raimille1) [inline source/includes/getting-started.md]: Add a comment here that this is after you finish a route. I know we want to make them more general purpose, but let's keep it simple for now.

File: source/index.html.md [hunk 3]
```diff
@@ -10,7 +10,7 @@ toc_footers:
   - <a href='https://foxtrotsystems.github.io/android-sdk-javadoc/'>Javadoc</a>
   - <a href='https://app.foxtrot.io/terms'>Terms of Service</a>
   - <a href='https://app.foxtrot.io/privacy'>Privacy Policy</a>
-  - Last Updated 2018-01-26
+  - Last Updated 2018-04-17
```
Comment (raimille1) [inline source/index.html.md]: 👀

File: source/includes/getting-started.md (hunk 2 above)
Comment (loganakamatsu) [reply source/includes/getting-started.md]: Wait you actually finish the route right below this... I'll clear it up though.

Comment (raimille1) [reply source/includes/getting-started.md]: ahh right right .. ok ok missed it sorry. ... and yeah it should be registered before finishing

**Comment by raimille1** [comment]
:shipit:

**Pull request merged**